cli/src/pcluster/validators/slurm_settings_validator.py

```diff
@@ -144,3 +144,36 @@ def _validate(self, custom_settings: List[Dict], include_file_url: str):
                 "under SlurmSettings.",
                 FailureLevel.ERROR,
             )
+
+
+class SlurmNodePrioritiesWarningValidator(Validator):
+    """
+    Slurm Node Weights Warning Validator.
+
+    This validator checks, within a queue, whether any dynamic nodes have lower node weights than any static
+    nodes and throws a warning if that's the case.
+    """
+
+    def _validate(self, queue_name: str, compute_resources: List[Dict]):
+        static_priorities = {cr.name: cr.static_node_priority for cr in compute_resources}
+        dynamic_priorities = {cr.name: cr.dynamic_node_priority for cr in compute_resources}
+
+        bad_static_priorities = {
+            key: value
+            for key, value in static_priorities.items()
+            if value in range(min(dynamic_priorities.values()), max(dynamic_priorities.values()) + 1)
```

Comment:
These functions are evaluated once for each `static priorities` and below once per `dynamic priority`.
Moreover we aren't interested in the full range but only in on boundary: `max_static` and `min_dynamic`
and we want to ensure that "all static are <= min_dynamic" and "all dynamic are >= max_static".

Reply:
Oh, you mean calculating first the boundaries of the ranges and using them "statically" in the if condition?

Honest doubt here: wouldn't Python's interpreter try to optimize something here? But I agree we can be stricter.

Reply:
Addressed, and we saved some vertical space in the code :)